modules/tinymce/src/plugins/table/test/ts/browser/ui/TableClassListButtonsTest.ts

```diff
@@ -0,0 +1,126 @@
+import { UiFinder, Waiter } from '@ephox/agar';
+import { afterEach, describe, it } from '@ephox/bedrock-client';
+import { TinyHooks, TinySelections, TinyUiActions } from '@ephox/mcagar';
+import { SugarBody } from '@ephox/sugar';
+import { assert } from 'chai';
+import Editor from 'tinymce/core/api/Editor';
+import { EditorEvent } from 'tinymce/core/api/util/EventDispatcher';
+import { TableModifiedEvent } from 'tinymce/plugins/table/api/Events';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.plugins.table.ui.TableClassListButtonsTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'table',
+    indent: false,
+    table_class_list: [
+      {
+        title: 'A',
+        value: 'a'
+      },
+      {
+        title: 'B',
+        value: 'b'
+      }
+    ],
+    toolbar: 'tableclass',
+    base_url: '/project/tinymce/js/tinymce',
+    setup: (editor: Editor) => {
+      editor.on('tablemodified', logEvent);
+    }
+  }, [ Plugin, Theme ], true);
+
+  let events: Array<EditorEvent<TableModifiedEvent>> = [];
+  const logEvent = (event: EditorEvent<TableModifiedEvent>) => {
+    events.push(event);
+  };
+
+  afterEach(() => {
+    events = [];
+
+    try { // Check if the menu is still open. If it is, close it.
+      const editor = hook.editor();
+      UiFinder.exists(SugarBody.body(), '.tox-menu');
+      TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+    } catch (e) {}
+  });
+
+  it('TINY-7476: Ensure that the checkmark appears for a single class', async () => {
+    const editor = hook.editor();
+
+    editor.setContent(
+      '<table>' +
+        '<tbody>' +
+          '<tr>' +
+            '<td>Filler</td>' +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+
+    const sugarContainer = SugarBody.body();
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+    await Waiter.pTryUntil('There should not be a checkmark', () => {
+      UiFinder.exists(sugarContainer, '.tox-menu');
+      UiFinder.exists(sugarContainer, '.tox-collection__item[aria-checked="false"]'); // No checked ones exist.
+      UiFinder.notExists(sugarContainer, '.tox-collection__item[aria-checked="true"]');
+    });
+
+    editor.execCommand('mceTableToggleClass', false, 'a');
+    assert.lengthOf(events, 1, 'Command executed successfully.');
+
+    // Cycle the menu to let the checkmark load.
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
```

Comment:
This is rather confusing... a better/clearer way to handle closing menus which is what's used elsewhere is to send the `escape` key event. I'd also recommend splitting these out into `openMenu` and `closeMenu` functions which will improve readability greatly. Another suggestion I'd have is to move the close part to immediately after we've asserted the menu state.

- [ ] Improve this please